nemo/utils/model_utils.py

```diff
@@ -0,0 +1,230 @@
+                return key
+
+    return None
+
+
+def parse_dataset_as_name(filepath: str) -> str:
+    """
+    Constructs a valid prefix-name from a provided file path.
+
+    Args:
+        filepath: str path to some valid data/manifest file.
+
+    Returns:
+        str prefix used to identify uniquely this data/manifest file.
+    """
+    filename = Path(filepath).stem
+
+    # cleanup name
+    filename = filename.replace('-', '_')
+
+    if 'manifest' in filename:
+        filename = filename.replace('manifest', '')
+
+    if '_' != filename[-1]:
+        filename = filename + '_'
+
+    return filename
+
+
+def resolve_validation_dataloaders(model: 'ModelPT'):
+    """
+    Helper method that operates on the ModelPT class to automatically support
+    multiple dataloaders for the validation set.
+
+    It does so by first resolving the path to one/more data files via `resolve_filepath_from_cfg()`.
+    If this resolution fails, it assumes the data loader is prepared to manually support / not support
+    multiple data loaders and simply calls the appropriate setup method.
+
+    If resolution succeeds:
+        Checks if provided path is to a single file or a list of files.
+        If a single file is provided, simply tags that file as such and loads it via the setup method.
+        If multiple files are provided:
+            Inject a new manifest path at index "i" into the resolved key.
+            Calls the appropriate setup method to set the data loader.
+            Collects the initialized data loader in a list and preserves it.
+            Once all data loaders are processed, assigns the list of loaded loaders to the ModelPT.
+            Finally assigns a list of unique names resolved from the file paths to the ModelPT.
+
+    Args:
+        model: ModelPT subclass, which requires >=1 Validation Dataloaders to be setup.
+    """
+    cfg = copy.deepcopy(model._cfg)
+    dataloaders = []
+
+    # process val_loss_idx
+    if hasattr(cfg.validation_ds, 'val_loss_idx'):
+        cfg = OmegaConf.to_container(cfg)
+        val_loss_idx = cfg['validation_ds'].pop('val_loss_idx')
+        cfg = OmegaConf.create(cfg)
+    else:
+        val_loss_idx = 0
+
+    # Set val_loss_idx
+    model._validation_loss_idx = val_loss_idx
+
+    ds_key = resolve_dataset_name_from_cfg(cfg.validation_ds)
+
+    if ds_key is None:
+        logging.debug(
+            "Could not resolve file path from provided config - {}. "
+            "Disabling support for multi-dataloaders.".format(cfg.validation_ds)
+        )
+
+        model.setup_validation_data(cfg.validation_ds)
+        return
+
+    ds_values = cfg.validation_ds[ds_key]
+
+    if type(ds_values) in (list, tuple, ListConfig):
+
+        for ds_value in ds_values:
+            cfg.validation_ds[ds_key] = ds_value
+            model.setup_validation_data(cfg.validation_ds)
+            dataloaders.append(model._validation_dl)
+
+        model._validation_dl = dataloaders
```

Comment:
This works with lightning? I'm surprised

Reply:
Lightning is after all at it's core just a python class object. So yep it works.